subsys/partition_manager/Kconfig.template.build_strategy

```diff
@@ -24,4 +24,12 @@ config $(module)_BUILD_STRATEGY_FROM_SOURCE
 	# Mandatory option when being built through add_child_image'
 	bool "Build from source"
 
+if "$(module)"="B0" && $(module)_BUILD_STRATEGY_FROM_SOURCE
+
+config $(module)_SOURCE_DIR
+	# Optional when being built through add_child_image'
+	string "$(module) source directory (relative path to the project directory)"
```

Comment:
This will prevent users from creating a Kconfig fragment like:
b0-fragment.conf
```
CONFIG_B0_SOURCE_DIR=/path/to/module/
```

and then do:
`cmake -DOVERLAY_CONFIG=b0-fragment.conf some/path/app1`
`cmake -DOVERLAY_CONFIG=b0-fragment.conf another/longer/path/app2`

In general, relative paths should not be used in Kconfig, as there is no common understanding regarding relative to what.

Reply:
That's a valid point, thanks! I will make it an absolute path